### PR TITLE
Allow MOTECH to only export or only import some values to/from OpenMRS

### PR DESCRIPTION
**Pull request by kaapstorm**
Feature flag: `openmrs_integration` / "Enable OpenMRS integration"

:blowfish:

### CODE REVIEW

**Comment by kaapstorm** [comment]
(Not sure why **dimagi/required-labels** is unhappy.)

**Review by gbova — APPROVED** [review]

File: corehq/motech/value_source.py [hunk 1]
```diff
@@ -52,6 +67,22 @@ def deserialize(self, external_value):
...
+        >>> value_source = ValueSource(direction=DIRECTION_BOTH)
+        >>> bool(value_source.check_direction(DIRECTION_EXPORT))
+        True
+
+        >>> value_source = ValueSource(direction=DIRECTION_IMPORT)
+        >>> bool(value_source.check_direction(DIRECTION_EXPORT))
+        False
+
+        """
+        return DIRECTIONS[self.direction] & DIRECTIONS[direction]
```
Comment (mkangia) [inline corehq/motech/value_source.py]: the logic with bytes is pretty cool. 
Though might be worth to just keep it simple 
`self.direction && direction == self.direction`
and get rid of 
```
DIRECTION_BOTH = None
DIRECTIONS = {
    DIRECTION_IMPORT: 0b01,
    DIRECTION_EXPORT: 0b10,
    DIRECTION_BOTH: 0b11,
}
```

**Review by mkangia — APPROVED** [review]

File: corehq/motech/value_source.py (hunk 1 above)
Comment (kaapstorm) [reply corehq/motech/value_source.py]: Yeah, that sounds fair. The bytes approach was based on what we did with the 99dots integration, but I agree this is simpler.

File: corehq/motech/value_source.py [hunk 2]
```diff
@@ -73,15 +73,15 @@ def check_direction(self, direction):
...
         True
 
         >>> value_source = ValueSource(direction=DIRECTION_IMPORT)
-        >>> bool(value_source.check_direction(DIRECTION_EXPORT))
+        >>> value_source.check_direction(DIRECTION_EXPORT)
         False
 
         """
-        return DIRECTIONS[self.direction] & DIRECTIONS[direction]
+        return not self.direction or direction == self.direction
```
Comment (mkangia) [inline corehq/motech/value_source.py]: this should be `and`, not `or`. Also, It would give an error when comparing with `None` i believe.

Comment (kaapstorm) [reply corehq/motech/value_source.py]: I'm not sure I understand.

`self.direction` is `None` when the integration should both import and export values, in which case this method should return `True` regardless of the value of `direction`. If `self.direction` is "out" then it must return `True` if `direction` is also "out", and likewise for "in". i.e:
```python
>>> other_direction = "in"
>>> self_direction = None
>>> not self_direction or other_direction == self_direction
True
>>> self_direction = "out"
>>> not self_direction or other_direction == self_direction
False
>>> self_direction = "in"
>>> not self_direction or other_direction == self_direction
True
```

Comment (mkangia) [reply corehq/motech/value_source.py]: oh yes. You are right.

**Pull request merged**